crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2790,6 +2809,12 @@ impl KnownFunction {
     }
 }
 
+#[salsa::interned]
+pub struct ModuleLiteralType<'db> {
+    pub containing_file: File,
```

Comment:
hmm, why do we need to store the `File` as a separate field on `ModuleLiteralType`, when the first field is a `Module`, and `Module` already stores the file?

The definition of `Module` is here -- it's just a thin wrapper around `ModuleInner`:

https://github.com/astral-sh/ruff/blob/c4df0a29ef81813b0f8274ba36064b60c0fbd265/crates/red_knot_python_semantic/src/module_resolver/module.rs#L9-L13

And `ModuleInner` is here:

https://github.com/astral-sh/ruff/blob/c4df0a29ef81813b0f8274ba36064b60c0fbd265/crates/red_knot_python_semantic/src/module_resolver/module.rs#L64-L70

The `Module` struct exposes a public getter method to retrieve the `File` of the module:

https://github.com/astral-sh/ruff/blob/c4df0a29ef81813b0f8274ba36064b60c0fbd265/crates/red_knot_python_semantic/src/module_resolver/module.rs#L37-L40

Reply:
It's not the same File. This is the file in which the import occurred, not the file of the module being imported. One of my comments recommends clarifying doc comments here.

Reply:
Ahh, thanks. I should have looked more closely -- my bad. Yes, I definitely support a better name for the field and more doc-comments, in that case! :-)